src/Microsoft.Diagnostics.Runtime/Interfaces/IClrRuntime.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace Microsoft.Diagnostics.Runtime.Interfaces
+{
+    public interface IClrRuntime : IDisposable
+    {
+        ImmutableArray<IClrAppDomain> AppDomains { get; }
+        IClrModule BaseClassLibrary { get; }
+        IClrInfo ClrInfo { get; }
+        IDataTarget DataTarget { get; }
+        ClrHeap Heap { get; }
+        bool IsThreadSafe { get; }
+        IClrAppDomain? SharedDomain { get; }
+        IClrAppDomain? SystemDomain { get; }
+        ImmutableArray<IClrThread> Threads { get; }
+
+        IEnumerable<ClrNativeHeapInfo> EnumerateClrNativeHeaps();
```

Comment:
Have an interface for ClrNativeHeapInfo?

Reply:
public constructor